packages/http-message-sig/src/parse.ts

```diff
@@ -19,19 +28,34 @@ function parseEntry(
   if (value.match(/^".*"$/)) return [key.trim(), value.slice(1, -1)];
   if (value.match(/^\d+$/)) return [key.trim(), parseInt(value)];
 
+  // TODO: this is restricted to components array. Per RFC9421, there could be more
   if (value.match(/^\(.*\)$/)) {
-    const arr = value
-      .slice(1, -1)
-      .split(/\s+/)
-      .map((entry) => entry.match(/^"(.*)"$/)?.[1] ?? parseInt(entry));
+    const arr = value.slice(1, -1).split(/\s+/);
+
+    const res = [];
+    for (const item of arr) {
+      const match = item.match(/^"(.*)"$/);
+      let toPush;
+      if (!match) {
+        toPush = parseInt(item);
+      } else if (match[1].includes('";key="')) {
+        toPush = {
+          key: match[1].split('";key="')[1],
```

Comment:
nit: this works if `key`  is the first parameter on the inner list, and it works for the specific case of `"signature-agent";key"...."` today because it's the _only_ parameter we require. 

But for cases of two or more keys, where order is not fixed, this will break. I think a proper parser of innerlist parameters for incoming requests should handle that.